JASP-Engine/JASP/R/principalcomponentanalysis.R

```diff
@@ -842,44 +872,59 @@ PrincipalComponentAnalysis <- function(dataset = NULL, options, perform = "run",
 
 ### Screeplot:
 
-.screePlot <- function(dataset, options, perform) {
+.screePlot <- function(dataset, options, perform, oldPlot = NULL, plotType) {
+
+  # After image can be wrote to file, the state system can be used to return unchanged screePlot
+  if (!is.null(oldPlot) && !identical(oldPlot[["data"]], "") && !is.null(oldPlot[["data"]])){
+    return(oldPlot)
+  }
   
+  # print("drawing screePlot...")
   screePlot <- list()
+  screePlot$title <- "Scree Plot"
+  screePlot$width <- options$plotWidthScreePlot
+  screePlot$height <- options$plotHeightScreePlot
+  screePlot$custom <- list(width="plotWidthScreePlot", height="plotHeightScreePlot")
   
-  #   if (perform == "run" && status$ready && !status$error && !is.null(model)) {
-  #    
-  if (perform == "run" && ncol(dataset) > 0 && nrow(dataset)> 0 && length(options$variables) > 1) {
-    
-    screePlot$title <- "Scree Plot"
-    screePlot$width <- options$plotWidthScreePlot
-    screePlot$height <- options$plotHeightScreePlot
-    screePlot$custom <- list(width="plotWidthScreePlot", height="plotHeightScreePlot")
-    
+  if (!is.null(dataset) && nrow(dataset)> 1 && length(options$variables) > 1) { 
     # Compute ev:
     
     # Compute ev:
     image <- .beginSaveImage()
     pa <- psych::fa.parallel(dataset)   
     .endSaveImage(image)
     
```

Comment:
@hummyhummy plot function to a common r file

Reply:
this is part of the function screenPlot(), which is a common function only in file principalcomponentanalysis.R